compiler/rustc_typeck/src/collect/type_of.rs

```diff
@@ -170,19 +171,47 @@ pub(super) fn opt_const_param_of(tcx: TyCtxt<'_>, def_id: LocalDefId) -> Option<
                         .position(|arg| arg.id() == hir_id)
                         .map(|index| (index, seg))
                 });
+            // FIXME(associated_const_equality): recursively search through the bindings instead
+            // of just top level.
+
             let (arg_index, segment) = match filtered {
                 None => {
-                    tcx.sess
-                        .delay_span_bug(tcx.def_span(def_id), "no arg matching AnonConst in path");
-                    return None;
+                    let binding_filtered = path
+                        .segments
+                        .iter()
+                        .filter_map(|seg| seg.args.map(|args| (args.bindings, seg)))
+                        .find_map(|(bindings, seg)| {
```

Comment:
you should be able to do the entire logic inside `find_map` by just doing `let bindings = seg.args?.bindings;` inside the `find_map`.